pyaedt/icepak.py

```
@@ -6051,3 +6051,93 @@ def assign_conducting_plate_with_conductance(self, obj_plate, boundary_name=None
+            return converted_value
+        except KeyError:
+            self.logger.info("Not a proper input unit")
+            return value
+
+    @pyaedt_function_handler()
+    def _create_cylinder(self, df, unit):
+        cylinders = []
+        for i in range(len(df)):
+            try:
+                name = self._remove_special_characters(df["name"][i])
+                radius = self._length_unit_conversion(df['radius'][i], input_units=unit)
+                height = self._length_unit_conversion(df['height'][i], input_units=unit)
+                cs_axis = self._length_unit_conversion(df['plane'][i], input_units=unit)
+                inner_radius = self._length_unit_conversion(df['iradius'][i], input_units=unit)
+                xc = self._length_unit_conversion(df['xc'][i], input_units=unit)
+                yc = self._length_unit_conversion(df['yc'][i], input_units=unit)
+                zc = self._length_unit_conversion(df['zc'][i], input_units=unit)
+                position = (xc, yc, zc)
+            except KeyError:
+                self.logger.info("The column names in the file do not match the expected names")
+                return False
+            try:
+                if inner_radius == 0:
+                    outer_cylinder = self.modeler.create_cylinder(cs_axis, position, radius, height, name=name)
+                elif radius > inner_radius:
+                    outer_cylinder = self.modeler.create_cylinder(cs_axis, position, radius, height, name=name)
+                    inner_cylinder = self.modeler.create_cylinder(cs_axis, position, inner_radius, height,
+                                                                  name=name + "_inner")
+                    outer_cylinder.subtract([inner_cylinder], keep_originals=False)
+                else:
+                    self.logger.info("Inner radius is greater than outer radius")
```

Should we raise an error if iradius>radius?